### PR TITLE
vim-patch:9.1.{0234-0250}: lots of filetypes not being recognized

### PR DESCRIPTION
**Pull request by clason**
Problem:  filetype: support for Intel HEX files is lacking
Solution: Add more file extensions that are typical for Intel HEX files
          (Wu, Zhenyu)

Reference: https://en.wikipedia.org/wiki/Intel_HEX

closes: vim/vim#14355

https://github.com/vim/vim/commit/e523dd9803ed62ea0657af8c85ab7bdfe80f4c53

Co-authored-by: Wu, Zhenyu <wuzhenyu@ustc.edu>

### CODE REVIEW

**Comment by clason** [comment]
first of a ~20 vim pr bomb; will add them to this PR as they're merged upstream.

File: runtime/lua/vim/filetype.lua [hunk 1]
```diff
@@ -2049,6 +2051,7 @@ local pattern = {
   ['XF86Config%-4.*'] = starsetf(detect.xfree86_v4, { priority = -math.huge + 1 }),
   ['XF86Config.*'] = starsetf(detect.xfree86_v3),
   ['%.zcompdump.*'] = starsetf('zsh'),
+  ['%.zsh_history'] = starsetf('zsh'),
```
Comment (zeertzjq) [inline runtime/lua/vim/filetype.lua:2060]: This should be in the `filename` table and should not use `starsetf()`.

Comment (clason) [reply runtime/lua/vim/filetype.lua:2060]: Why this one and not the one above?

Comment (clason) [reply runtime/lua/vim/filetype.lua:2060]: (`starsetf`, that is)

Comment (zeertzjq) [reply runtime/lua/vim/filetype.lua:2060]: 1. It's not used in Vim.
2.
```vim
" Function used for patterns that end in a star: don't set the filetype if the
" file name matches ft_ignore_pat.
" When using this, the entry should probably be further down below with the
" other StarSetf() calls.
func s:StarSetf(ft)
  if expand("<amatch>") !~ g:ft_ignore_pat
    exe 'setf ' . a:ft
  endif
endfunc
```

**Pull request merged**